### ISSUE TITLE
Grab log levels from environment (LOGLEVEL / LOGLEVEL_label)

### ISSUE THREAD

**Issue by helje5** [apple/swift-log#110]
It would be great if the default `StreamLogHandler` would honour LOGLEVEL environment variables.

To change the loglevel I currently have to do this dance, right?:
```swift
LoggingSystem.bootstrap { label in
  var handler = StreamLogHandler.standardOutput(label: label)
  handler.logLevel = .trace
  return handler
}
```

Because `StreamLogHandler` hardcodes its level to `.info`:
```swift
public var logLevel: Logger.Level = .info
```
 It would be nice if that would at least default to the setting of the `LOGLEVEL` environment variable, ideally w/ supporting `LOGLEVEL_$label` as well.
Essentially:
```swift
public var logLevel: Logger.Level = {
  let env = ProcessInfo.processInfo.environment
  return Logger.Level(rawValue: "LOGLEVEL_" + label)
         ?? Logger.Level(rawValue: "LOGLEVEL")
         ?? .info
}()
```
Maybe w/ a little more caching.

**Comment by ianpartridge** [comment]
No objection from me to the principle of the feature, although I would suggest we make the name a bit less generic to avoid collisions with any other software that might be in the picture - perhaps `SWIFT_LOG_LOGLEVEL`.

I bet @weissi will be against this though as he wants to keep the default logger super minimal and basic.

**Comment by helje5** [comment]
Just because he doesn't use it! :-) `SWIFT_LOG_LOGLEVEL` as yet another layer sounds OK, but it should still fallback to LOGLEVEL IMO.

**Comment by Yasumoto** [comment]
This would be quite handy for on-the-fly tweaking. I get the reasoning for keeping the default implementation simple but think there are enough "best practices" we should encourage that I'd like to see something that encapsulates things like this 🤔

**Comment by ktoso** [comment]
This is another case of an issue that is the growing features in the `StreamLogHandler` in a package that was intended to be an "API only" package 😟 To be honest tend to thing it would have been much better if this package never provided any implementation _at all_ I keep thinking (as we did with http://reactive-streams.org)...

I could be convinced that this is small enough to consider "this is small enough", but overall I'm worried we're going down the wrong path growing features here, rather than collaborating on a full blown stream (stdout and/or file) `LogHandler`, with configuration, output formatting etc. I'm also worried about setting too much precedent about: using "mainly" env as source of logging/config (does not scale to actually configuring many loggers) and naming convention here, though perhaps these are a lesser worry here.

I would _love_ to see us come up and collaborate with a "_serious_" log handler implementation that would supersede all of the ones that are provided in here. Should we set something like that up and try to collaborate on getting a "real world" nice stream handler as a separate package?

TL;DR; IMHO: This library really should not become _the best_ stream/stdout/file logger. It should be just the bare minimum APIs and be pretty limited such that it is well apparent that using a different "serious" backend is recommended, and we should make such one happen -- who's interested?

**Comment by Yasumoto** [comment]
Just this afternoon (I’m _very_ slow after the holidays 😅 ) I came across another Issue where you explained this rationale; and that finally clicked for me. Totally down for separating this as an “API package” vs another “default” one.

I think I’d just love there to be a default robust one! I don’t have cycles to lead the charge (unless no one else does), but happy to contribute!

**Comment by weissi** [comment]
Sorry for the delay, I agree with @ktoso . I really don't want to encourage users to use the default backend so I'm -1 on the proposed change here. @helje5 why don't you make a new backend that literally starts with a copy & pasted version of the `StreamLogHandler`?

**Comment by helje5** [comment]
Yeah, I just replaced my own logging implementation with this one, my expectations have been a little wrong I guess :-) I can do my own.

TBH I'd very much prefer if swift-log itself comes with a reasonable default implementation(s). But maybe what's reasonable is too hard to agree on :-)

**Comment by timetraveler90** [comment]
Is there any workaround to avoid hardcoding the level like this:

`public var logLevel: Logger.Level = .info`

I'd prefer to have something like this maybe, but it's only my personal preferences:

`public var logLevels: Logger.Level = [.info, .debug, .critical, .error]`

**Comment by weissi** [comment]
I'm going to close this because this is an API package. More thoughts here: https://forums.swift.org/t/thoughts-on-why-does-swiftlog-not-have-multiple-log-destinations-formatters-colouring-some-other-feature/37507
